htsget-search/src/storage/local.rs

```diff
@@ -139,7 +140,70 @@ mod tests {
     });
   }
 
-  // TODO add tests for `LocalStorage::url`
+  #[test]
+  fn url_of_non_existing_key() {
+    with_local_storage(|storage| {
+      let result = storage.url("non-existing-key", UrlOptions::default());
+      assert_eq!(
+        result,
+        Err(StorageError::InvalidKey("non-existing-key".to_string()))
+      );
+    });
+  }
+
+  #[test]
+  fn url_of_folder() {
+    with_local_storage(|storage| {
+      let result = storage.url("folder", UrlOptions::default());
+      assert_eq!(result, Err(StorageError::NotFound("folder".to_string())));
+    });
+  }
+
+  #[test]
+  fn url_with_forbidden_path() {
+    with_local_storage(|storage| {
+      let result = storage.url("folder/../../passwords", UrlOptions::default());
+      assert_eq!(
+        result,
+        Err(StorageError::InvalidKey(
+          "folder/../../passwords".to_string()
+        ))
+      );
+    });
+  }
+
+  #[test]
+  fn url_of_existing_key() {
+    with_local_storage(|storage| {
+      let result = storage.url("folder/../key1", UrlOptions::default());
+      let expected = Url::new(format!(
+        "file://{}",
+        storage.base_path().join("key1").to_string_lossy()
```

Comment:
Why `.to_string_lossy()`? Is there a better way to handle this? Perhaps assuming ascii-only characters in the URL instead of utf-8?

Reply:
I did this in the test because it is also [how it works in the implementation](https://github.com/chris-zen/htsget-mvp/blob/066ed6e3648d59938c1bd98556ebfa25dbfb3cf4/htsget-search/src/storage/local.rs#L75). That said I think it is probably okay  to use `.to_string_lossy()`. By that point we know that the path we use to construct the URL exists in the machine and the machine runs linux. Unless I am mistaken linux uses UTF8 to encode the paths so it should be valid UTF8. If we used `to_str`the result would be exactly the same, but this way we don't have to deal with the `Option`.  

I am not sure whether you can use UTF8 in HTTPS requests, but if it is possible and we wanted to restrict our server to only accept ASCII in the request it should probably be done earlier rather than after checking if the file exists in the storage

Reply:
it all started from lazyness to handle the way strings can be obtained from a Path (which is quite entangled), and assume that paths will be compatible with UTF-8 (am I being naive?). If you have a better proposal I'm all ears.
Anyway, for a test is more than fine, because we are in control on what strings to use, but I'm more interested for a solution in the logic side.